### DISCUSSION TITLE
How to clear the (people)picker control when items are added manually

### DISCUSSION THREAD

**Discussion by rbarten** [microsoft/powercat-creator-kit#265] [Q&A] [answered]
When I enter values in for example the people picker and then press a button to submit, I want to clear the values which were added to the picker control.
Currently I don't see an option to do that. For example Reset([control]) does work for a DatePicker but does not work for a People or 'normal' picker. Any ideas?

**Comment by yaseralhosani** [comment, marked as answer]
Hi @rbarten,

 The [Issues 236](https://github.com/microsoft/powercat-creator-kit/issues/236) has been raised. This is a temporary solution that I found.

**Button.OnSelect**
It would be best to replace User() with functions like AddColumns, Filter...etc. Basically, you need to add items to the collection and then set it to blank. You will need 2 Collections if 1 collection is not working, 
```
ClearCollect(
    PreSelectedUserCollection,
    User()
);
ClearCollect(
    PreSelectedUserCollection,
    Blank()
);
```
Only if you are using PreSelectedUserCollection, set the column name from the collection in  PeoplePicker.PersonaKey, PeoplePicker.PersonaName...etc

**PeoplePicker.Items**
This is only for preselected or leaves blank.
```
PreSelectedUserCollection
```

**PeoplePicker.OnSearch**
Search items
```
ClearCollect(
    UserCollection,
    AddColumns(
        Filter(
            'AAD Users',
            ThisRecord.'AAD User Account Enabled' = 'AAD User Account Enabled (AAD Users)'.Yes
        ),
        "UPN",
        ThisRecord.'User Principal Name',
        "DisplayName",
        ThisRecord.'Display Name
    )
)
```

**PeoplePicker.Suggestions_Items**
```
Search(
    UserCollection,
    Self.SearchText,
    "DisplayName",
    "UPN"
)
```
Only if you are using UserCollection, set the column name from the collection in PeoplePicker.SuggestionKey, PeoplePicker.SuggestionName...etc

If you still have problems, please let me know.

**Comment by nunon** [comment]
Exactly the same issue.

Items property:

`Blank()`


OnSearch property of people picker:
`
ClearCollect(
    colUserCollectionNovosUsers,
        AddColumns(
            Filter(
                UtilizadoresdoOffice365.SearchUser(
                    {
                        searchTerm: Self.SearchText,
                        top: 500
                    }
                ),
                !(Mail in Self.SelectedPeople.PersonaKey) &&
                !(Mail in Utilizadores_1.Title)                                 
            
            ),
            "SuggestionImgUrl",
            Substitute(
                JSON(
                    UtilizadoresdoOffice365.UserPhotoV2(Id),
                    JSONFormat.IncludeBinaryData
                ),
                """",
                ""
            ),
            "SuggestionKey",
            Mail,
            "SuggestionName",
            DisplayName,
            "SuggestionRole",
            JobTitle,
            "SuggestionPresence",
            "away"
        )
    )`

Suggestions_Items property:
`colUserCollectionNovosUsers`

How to clear the people picker box after the form is submitted??

There should be a method, but I've tried:
`Clear(colUserCollectionNovosUsers);          
ClearCollect(colUserCollectionNovosUsers, Blank());`

But, it doesn't work.

**Comment by bcobra** [comment]
Will there be any updates to the project? For Example, the people picker is still in preview and as far as I know there is no modern people picker control. The modern combo box seems to fall short.
Thanks
